pkgs/servers/http/apache-httpd/2.4.nix

```diff
@@ -39,6 +39,7 @@ stdenv.mkDerivation rec {
   prePatch = ''
     sed -i config.layout -e "s|installbuilddir:.*|installbuilddir: $dev/share/build|"
     sed -i support/apachectl.in -e 's|@LYNX_PATH@|${lynx}/bin/lynx|'
+    sed -i support/apachectl.in -e 's|$HTTPD -t|$HTTPD -t -f /etc/httpd/httpd.conf|'
```

Comment:
Why don't you patchin  the path of `${httpdConf}` here instead of the symlink? Then the symlink becomes unnecessary.

Reply:
The path of the conf file is determined in the module which depends on the package. We can't have the package depend on the module when the module depends on the package.

Reply:
The package could take that path as an argument (defaulting to `/etc/httpd/httpd.conf`) and then the module would pass the appropriate path to the packet.

Reply:
@peti the generated `httpd.conf` references the package itself, so if the package is modified that means the path in the nix store to `httpd.conf` is modified. I'm not a functional programmer, but even given the lazy semantics of `nix` I don't think that works... :man_shrugging:

Reply:
Oh, right. That would be an infinite recursion. :-(

I didn't realize that the generated `httpd.conf` would need to know the path of the package, but now that I think about it, I see why it might.

OK then, looks like the impure global path is the easiest solution for the time being.